internal/model/config.go

```diff
@@ -35,55 +35,53 @@ func NewConfig(options ...Option) *Config {
 // Option is an option you can pass to initialize minivpn.
 type Option func(config *Config)
 
-// WithConfigFile configures OpenVPNOptions parsed from the given file.
-func WithConfigFile(configPath string) Option {
-	return func(config *Config) {
-		openvpnOpts, err := ReadConfigFile(configPath)
-		runtimex.PanicOnError(err, "cannot parse config file")
-		runtimex.PanicIfFalse(openvpnOpts.HasAuthInfo(), "missing auth info")
-		config.openvpnOptions = openvpnOpts
-	}
-}
-
 // WithLogger configures the passed [Logger].
 func WithLogger(logger Logger) Option {
 	return func(config *Config) {
 		config.logger = logger
 	}
 }
 
+// Logger returns the configured logger.
+func (c *Config) Logger() Logger {
+	return c.logger
+}
+
 // WithHandshakeTracer configures the passed [HandshakeTracer].
 func WithHandshakeTracer(tracer HandshakeTracer) Option {
 	return func(config *Config) {
 		config.tracer = tracer
 	}
 }
 
-// Logger returns the configured logger.
-func (c *Config) Logger() Logger {
-	return c.logger
-}
-
 // Tracer returns the handshake tracer.
 func (c *Config) Tracer() HandshakeTracer {
 	return c.tracer
 }
 
-// OpenVPNOptions returns the configured openvpn options.
-func (c *Config) OpenVPNOptions() *OpenVPNOptions {
-	return c.openvpnOptions
+// WithConfigFile configures OpenVPNOptions parsed from the given file.
+func WithConfigFile(configPath string) Option {
+	return func(config *Config) {
+		openvpnOpts, err := ReadConfigFile(configPath)
```

Comment:
I am not super convinced by panicking here. Ideally, given that this is a library function, I suppose it would instead be better to return an error to the caller. A strategy to do this could be the following:

1. you redefine `Option` as `func(*Config) error`

2. when you are invoking the sequence of functions you in `NewConfig`, you also check for errors

Reply:
true, I agree that panic is not a good thing to do here. I got too used to add a few asserts for incorrect values, but this is public-facing API.

I think I would prefer to move this to a new issue to work after merging #69, since I've split and moved the code dealing with config.